components/sup/src/manager/service_updater/rolling_update_worker.rs

```diff
@@ -70,9 +70,9 @@ impl RollingUpdateWorker {
                 // leader dies and we are promoted to update leader.
                 match self.follower_wait_for_turn().await {
                     FollowerWaitForTurn::PromotedToLeader => self.leader_role().await,
-                    FollowerWaitForTurn::UpdateTo(maybe_new_ident) => {
+                    FollowerWaitForTurn::UpdateTo(new_ident) => {
                         // Update to the package we were instructed to
-                        self.package_update_worker.run(maybe_new_ident).await
+                        self.package_update_worker.run(Some(new_ident)).await
```

Comment:
I totally get why `#run` takes an `Option` (thanks for the clear documentation!). However, when reading the call sites, it is not quite as clear as it could be. Casually reading this, I immediately think "OK, so this means I could call `package_update_worker.run(None)`... but what does _that_ mean?" It's a variation on how I feel about seeing functions that take boolean arguments (e.g. `do_something(true, false)`).

It could make this more readable to have some simple facade calls that express the intention more clearly... maybe something like `package_update_worker.update_to(new_ident)` and `package_update_worker.update_to_latest()`, or something similar.

(That's not a blocker, though... just a suggestion!)

Reply:
That is much better, thanks! Addressed in [dea222](https://github.com/habitat-sh/habitat/pull/7455/commits/dea222117e8df49eb85583c59171668d931eb4ec)